module/zcommon/zfs_namecheck.c

```diff
@@ -417,7 +417,6 @@ pool_namecheck(const char *pool, namecheck_err_t *why, char *what)
 	return (0);
 }
 
-#if defined(_KERNEL)
```

Comment:
do not remove kernel guard where we can use shared files on kernel side builds and in userland.
for example (on DilOS): we are building zfs module files in userland with libzpool and should avoid kernel specific changes.

Reply:
The files are shared for libzpool here, too. The kernel guard isn't needed when you have done as you suggest above
```
#define ZFS_MODULE_PARAM(...)
#define EXPORT_SYMBOL(...)
```
Effectively the kernel guard has been refactored to consolidate the duplicated code and prevent mistakes. IMO that is an improvement.

Reply:
probably, add a comment about this section visibility for kernel space instead of of _KERNEL guard can help too.
i like and more understand _KERNEL definition in code to be sure that this code for kernel :)
but it is my own preference.

Reply:
@ikozhukhov that's really a matter of opinion. The FreeBSD in-tree port is littered with #ifdef Illumos with large swaths of illumos specific code which I find to to really obscure understanding. Most of us would like to see the ifdefs eliminated to the fullest extent possible.

Reply:
well, i think, in first iteration - it's fine. we can try to play with different platforms: you can try to play on FreeBSD, i'll try to play on DilOS - and we can see how it can be useful. Linux side will be checked by CI.
i'd like to say: LGTM and ship it!

Reply:
Thanks!